frontend/src/pages/project/FetchProjects.tsx

```diff
@@ -1,114 +1,125 @@
+                assignment_file: project_item.data.assignment_file,
+                deadline: new Date(d[1]),
+                deadline_description: d[0],
+                course_id: Number(project_item.data.course_id),
+                visible_for_students: Boolean(
+                  project_item.data.visible_for_students
+                ),
+                archived: Boolean(project_item.data.archived),
+                test_path: project_item.data.test_path,
+                script_name: project_item.data.script_name,
+                regex_expressions: project_item.data.regex_expressions,
+                short_submission: latest_submission,
+                course: course,
+              };
+            });
+          }
+          // contains no dealine:
+          return [
+            {
               project_id: project_id,
               title: project_item.data.title,
               description: project_item.data.description,
               assignment_file: project_item.data.assignment_file,
-              deadline: new Date(d[1]),
-              deadline_description: d[0],
+              deadline: undefined,
+              deadline_description: undefined,
               course_id: Number(project_item.data.course_id),
-              visible_for_students: Boolean(project_item.data.visible_for_students),
+              visible_for_students: Boolean(
+                project_item.data.visible_for_students
+              ),
               archived: Boolean(project_item.data.archived),
               test_path: project_item.data.test_path,
               script_name: project_item.data.script_name,
               regex_expressions: project_item.data.regex_expressions,
               short_submission: latest_submission,
-              course: course
-            }
-          })
+              course: course,
+            },
+          ];
+        } catch (e) {
+          return [];
         }
-        // contains no dealine:
-        return [{
-          project_id: project_id,
-          title: project_item.data.title,
-          description: project_item.data.description,
-          assignment_file: project_item.data.assignment_file,
-          deadline: undefined,
-          deadline_description: undefined,
-          course_id: Number(project_item.data.course_id),
-          visible_for_students: Boolean(project_item.data.visible_for_students),
-          archived: Boolean(project_item.data.archived),
-          test_path: project_item.data.test_path,
-          script_name: project_item.data.script_name,
-          regex_expressions: project_item.data.regex_expressions,
-          short_submission: latest_submission,
-          course: course
-        }]
-
-      }catch (e){
-        return []
-      }
-    }
-
-    ));
-    formattedData = formattedData.flat()
-    return formattedData
+      })
+    );
+    formattedData = formattedData.flat();
+    return formattedData;
   } catch (e) {
```

Comment:
error e is now thrown so can be removed i think?

Reply:
[e82d4fe](https://github.com/SELab-2/UGent-3/pull/226/commits/e82d4fef2cdf445ed820019ac6bc2f7382fb21d3)